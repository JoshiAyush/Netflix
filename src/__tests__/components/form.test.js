import React from "react";
import { render } from "@testing-library/react";

import { FormContainer } from "../../components/index.js";

jest.mock("react-router-dom");

describe("<Form />", () => {
  it("renders the <Form /> with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <FormContainer>
        <FormContainer.Inner>
          <FormContainer.Title>Sign In Now</FormContainer.Title>

          <FormContainer.Base>
            <FormContainer.Input
              type="email"
              placeholder="Email Address"
              autoComplete="on"
            />

            <FormContainer.Input
              type="password"
              placeholder="Password"
              autoComplete="on"
            />

            <FormContainer.Submit type="submit" disabled>
              Sign In
            </FormContainer.Submit>
          </FormContainer.Base>

          <FormContainer.Text>New to Netflix?</FormContainer.Text>

          <FormContainer.TextSmall>
            This page is protected by Google reCAPTCHA to ensure you're not a
            bot. Learn more.
          </FormContainer.TextSmall>
        </FormContainer.Inner>
      </FormContainer>
    );

    expect(
      getByText(
        "This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more."
      )
    ).toBeTruthy();
    expect(getByText("Sign In Now")).toBeTruthy();
    expect(getByText("Sign In").disabled).toBeTruthy();
    expect(getByPlaceholderText("Email Address")).toBeTruthy();
    expect(getByPlaceholderText("Password")).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
