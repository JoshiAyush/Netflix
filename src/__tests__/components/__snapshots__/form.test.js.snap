// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<div
  class="style__Container-sc-51a5pf-0 gwxlzA"
>
  <div
    class="style__Inner-sc-51a5pf-1 jVmATQ"
  >
    <h1
      class="style__Title-sc-51a5pf-8 ftUewI"
    >
      Sign In Now
    </h1>
    <form
      class="style__Base-sc-51a5pf-6 kcVkXA"
    >
      <input
        autocomplete="on"
        class="style__Input-sc-51a5pf-11 cjBdKU"
        placeholder="Email Address"
        type="email"
      />
      <input
        autocomplete="on"
        class="style__Input-sc-51a5pf-11 cjBdKU"
        placeholder="Password"
        type="password"
      />
      <button
        class="style__Submit-sc-51a5pf-12 ejtinN"
        disabled=""
        type="submit"
      >
        Sign In
      </button>
    </form>
    <p
      class="style__Text-sc-51a5pf-7 kbrTat"
    >
      New to Netflix?
    </p>
    <p
      class="style__TextSmall-sc-51a5pf-9 eWkTIf"
    >
      This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
    </p>
  </div>
</div>
`;
